### PR TITLE
Add auto applying of dark theme

### PR DESCRIPTION
**Pull request by liubich**

### CODE REVIEW

**Comment by liubich** [comment]
I'm not sure, that the plugin approach is the best way in this case.

File: src/store/mutationTypes.js [hunk 1]
```diff
@@ -3,3 +3,4 @@ export const SAVE_CURRENT_WEATHER = 'SAVE_CURRENT_WEATHER';
 export const SAVE_ERROR_DESC = 'SAVE_ERROR_DESC';
 export const SAVE_CURRENT_POSITION_DATA = 'SAVE_CURRENT_POSITION_DATA';
 export const SAVE_HOURLY_FORECAST = 'SAVE_HOURLY_FORECAST';
+export const SAVE_PREFER_DARK_THEME = 'SAVE_PREFER_DARK_THEME';
```
Comment (probil) [inline src/store/mutationTypes.js]: [Optional] Call it `SAVE_PREFERRED_THEME` (and pass `dark` as an argument later). In this case it's much easier to add new theme

File: src/store/themeModule.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import * as mutationTypes from './mutationTypes';
+// eslint-disable-next-line import/prefer-default-export
```
Comment (probil) [inline src/store/themeModule.js]: There is no need to disable `import/prefer-default-export`:
```js
export default  {
  state: {}
  // ...
}
```
later in `store/index.js`
```js
import themeModule from './themeModule';
```
---
Or as an alternative you can import everything,e .g.

```js
export const state  = {
  preferDarkTheme: null,
}

export const mutations = {
  [mutationTypes.SAVE_PREFER_DARK_THEME](state, preferDarkTheme) {
      state.preferDarkTheme = preferDarkTheme;
  },
};
```
later in `store/index.js`
```js
import * as themeModule from './themeModule';
```

File: public/index.html [hunk 3]
```diff
@@ -36,6 +36,7 @@
       sizes="180x180"
       href="<%= BASE_URL %>img/icons/apple-touch-icon.png"
     />
+    <link rel="stylesheet" href="themes.css">
```
Comment (probil) [inline public/index.html]: What is the reason of having dedicated stylesheet just for themes?

Screenshot 2020-01-08 at 10 15 00

Just add one more `<style></style>` block to the app component.

File: src/store/darkThemePlugin.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+import * as mutationTypes from './mutationTypes';
+
+export default function darkThemePlugin({ commit }) {
+  if (!matchMedia) return;
+  const preferDarkTheme = matchMedia('(prefers-color-scheme: dark)');
+
+  function savePreferDarkThemeToStore(mql) {
```
Comment (probil) [inline src/store/darkThemePlugin.js]: [optional] you can move function to the bottom this way
```js
if (!matchMedia) return;
const preferDarkTheme = matchMedia('(prefers-color-scheme: dark)');
savePreferDarkThemeToStore(preferDarkTheme);
preferDarkTheme.addListener(savePreferDarkThemeToStore);

function savePreferDarkThemeToStore(mql) {
 commit(mutationTypes.SAVE_PREFERRED_THEME, mql.matches ? 'dark' : 'light');
}
```

File: src/store/darkThemePlugin.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    commit(mutationTypes.SAVE_PREFERRED_THEME, mql.matches ? 'dark' : 'light');
```
Comment (probil) [inline src/store/darkThemePlugin.js]: [optional] I would use mutation this way
```js
import { SAVE_PREFERRED_THEME } from './mutationTypes';
// later
commit(SAVE_PREFERRED_THEME, mql.matches ? 'dark' : 'light');
```

**Pull request merged**